packages/headless/doc-parser/src/actions-resolver.ts

```diff
@@ -0,0 +1,225 @@
+import {
+  ApiEntryPoint,
+  ApiInterface,
+  ApiItem,
+  ApiItemKind,
+  ApiNamespace,
+  ApiVariable,
+} from '@microsoft/api-extractor-model';
+import {DocNode, DocParamBlock} from '@microsoft/tsdoc';
+import {findApi} from './api-finder';
+import {AnyEntity, Entity, ObjEntity} from './entity';
+import {buildObjEntity} from './entity-builder';
+import {resolveInterfaceMembers} from './interface-resolver';
+import {emitAsTsDoc} from './tsdoc-emitter';
+
+interface ActionsNamespace {
+  name: string;
+  members: Action[];
+}
+
+enum ActionKind {
+  Normal = 'normalAction',
+  Analytics = 'analyticsAction',
+}
+
+interface BaseAction {
+  name: string;
+  desc: string;
+  parameter: AnyEntity | null;
+  actionKind: ActionKind;
+}
+
+enum AnalyticsKind {
+  Search = 'search',
+  Click = 'click',
+  Custom = 'custom',
+}
+
+interface AnalyticsAction extends BaseAction {
+  actionKind: ActionKind.Analytics;
+  analyticsKind: AnalyticsKind;
+}
+
+interface NormalAction extends BaseAction {
+  actionKind: ActionKind.Normal;
+}
+
+type Action = NormalAction | AnalyticsAction;
+
+export function resolveActionNamespace(
+  entry: ApiEntryPoint,
+  name: string
+): ActionsNamespace {
+  const namespace = findApi(entry, name) as ApiNamespace;
+  return {
+    name: namespace.displayName,
+    members: namespace.members
+      .filter((member) => member.kind === ApiItemKind.Variable)
```

Comment:
Unfortunately, I found that every member of a namespace is currently resolved as a variable due to the way we export actions.
<details>
<summary>example:</summary>

```ts
import {
  buildSearchRequest as buildSearchRequestAlias,
  executeSearch as executeSearchAlias,
  fetchMoreResults as fetchMoreResultsAlias,
} from './search/search-actions';
export namespace SearchActions {
  export const buildSearchRequest = buildSearchRequestAlias;
  /**
   * Executes a search query.
   * @param analyticsAction - The analytics action to log after a successful query.
   */
  export const executeSearch = executeSearchAlias;
  /**
   * Executes a search query and appends the results after the existing ones.
   */
  export const fetchMoreResults = fetchMoreResultsAlias;
}
```

</details>

This is why resolving the parameter type of an action is more complicated than the parameter type of a normal function.

A way around this would be to use the `export import` statement
<details>
<summary>example:</summary>

```ts
import * as _SearchActions from './search/search-actions';

export namespace SearchActions {
  export import buildSearchRequest = _SearchActions.buildSearchRequest;
  export import executeSearch = _SearchActions.executeSearch;
  export import fetchMoreResults = _SearchActions.fetchMoreResults;
}
```

</details>

Unfortunately, this expression isn't supported by `rollup-plugin-dts`, which we currently need for the api extractor to parse documentation.

I experimented with replacing `rollup-plugin-dts` with other plugins, but found that this would require either creating our own rollup plugin using [dts-bundle](https://www.npmjs.com/package/dts-bundle) or using a [different Typescript bundler](https://github.com/wessberg/rollup-plugin-ts) which is much less popular.

Instead of replacing `rollup-plugin-dts`, we could cope with the issues that using a variable brings and possibly make some improvements once https://github.com/microsoft/rushstack/pull/1916 is merged.